src/plugins/intel_cpu/tests/unit/gemm_api_test.cpp

```diff
@@ -7,8 +7,23 @@
 #include "mlas/sgemm.hpp"
 
 // This test is used to test whether mlas gemm lib compiles successfully
-TEST(GemmTests, getPackedSize) {
+TEST(MLASGemmTests, getPackedSize) {
     int N = 51864;
     int K = 384;
     ASSERT_NO_THROW(ov::intel_cpu::mlas_sgemm_pack_get_size(N, K));
+}
+// Test mlas thread partition with even/odd threads
+TEST(MLASGemmTests, simpleGemm) {
+    size_t M = 33;
+    size_t N = 32;
+    size_t K = 33;
+    std::vector<float> a_data(M * K, (1.0f/33));
+    std::vector<float> b_data(K * N, 4.0f);
+    std::vector<float> c_data(M * N, 0.0f);
+    ASSERT_NO_THROW(
+        ov::intel_cpu::
+            mlas_sgemm("N", "T", M, N, K, 1.0f, a_data.data(), K, b_data.data(), N, 0.0f, c_data.data(), N, 3));
```

Comment:
@zhangYiIntel is the “3”  in `mlas_sgemm("N", "T", M, N, K, 1.0f, a_data.data(), K, b_data.data(), N, 0.0f, c_data.data(), N, 3));`  the target thread number, maybe 33 or 34, because 33 or 34 is great than 32

Reply:
The testing machint in CI doesn't always has such more threads. It's hard to set 33 directly.

Reply:
@zhangYiIntel will the test case failed or crash if the target threads number is greater than the CPU threads?

Reply:
It may not crash, but actually the corner case we meet is caused by both threads num, the L2 cacheSize and data size. That's hard to find a very accurate test to simulate.

Reply:
from the comment in the code, 
` // Ensure that data used fits L2 cache. M_BLK = (L2/sizeof(float) - N * K ) / (K + N), ThreadCountM = M / M_BLK`
if we want to use the test case to cover the corner case, we should set M=128 and  M_BLK>128, then ThreadCountM < 1,
and for example, the L2 cache of My CPU is 512K, `M_BLK = (512*1024/8-32*32)(32+32) >128`, I am not sure my calculation is right, maybe wrong, But because most L2 cache of CPU is great than 512k now, so maybe we can find a good test case to cover it if it do not crash when target threads number is greater than the CPU threads

Reply:
Thanks for your advice, the test case is now constructed dynamically according to runtime threads & L2 cache size